conda/plan.py

```diff
@@ -689,33 +691,55 @@ def get_actions_for_dists(dists_for_prefix, only_names, index, force, always_cop
 
 
 def augment_specs(prefix, specs, pinned=True):
-    # get conda-meta/pinned
+    """
+    Include additional specs for conda and (optionally) pinned packages.
+
+    Parameters
+    ----------
+    prefix : str
+        Environment prefix.
+    specs : list of MatchSpec
+        List of package specifications to augment.
+    pinned : bool, optional
+        Optionally include pinned specs for the current environment.
+
+    Returns
+    -------
+    augmented_specs : list of MatchSpec
+       List of augmented package specifications.
+    """
+    specs = list(specs)
+
+    # Get conda-meta/pinned
     if pinned:
         pinned_specs = get_pinned_specs(prefix)
         log.debug("Pinned specs=%s", pinned_specs)
-        specs += [MatchSpec(spec) for spec in pinned_specs]
+        specs += list(pinned_specs)
```

Comment:
You can use `specs.extend(pinned_specs)` here to avoid creating a new list object that gets immediately thrown away.